substrate/frame/revive/src/lib.rs

```diff
@@ -267,6 +267,10 @@ pub mod pallet {
 			BlockNumberFor<Self>,
 		>;
 
+		/// The weigher used to calculate XCM execution costs
+		#[pallet::no_default_bounds]
+		type XcmWeigher: xcm_executor::traits::WeightBounds<<Self as frame_system::Config>::RuntimeCall>;
```

Comment:
Can't we add this bound to the existing `Xcm` associated type? I think its all implemented by `pallet_xcm` anyways.

Reply:
Since we moving the precompile anyways would it make sense to make changes in `pallet_xcm` to achieve this?

Reply:
The right thing to do would be to define the pre-compile in pallet_xcm.